src/nvim/getchar.c

```diff
@@ -2494,7 +2489,6 @@ fix_input_buffer (
   for (i = len; --i >= 0; ++p) {
     if (p[0] == NUL
         || (p[0] == K_SPECIAL
-          && !script
           && (i < 2 || p[1] != KS_EXTRA))) {
```

Comment:
Also in the original patch this comment on line 2493/2488 was removed:

    // Don't replace K_SPECIAL when reading a script file.